### PR TITLE
[Docs] Add Multi-scale training and testing

### PR DESCRIPTION
**Pull request by hhaAndroid**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Resolved https://github.com/open-mmlab/mmyolo/issues/629

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMClassification.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by PeterH0323 — APPROVED** [review]
LGTM

File: docs/zh_cn/common_usage/ms_training_testing.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# 多尺度训练和测试
+
+## 多尺度训练
+
+MMYOLO 中目前支持了主流的 YOLOv5、YOLOv6、YOLOv7、YOLOv8 和 RTMDet 等算法，其默认配置均为单尺度 640x640 训练。 在 MM 系列开源库中常用的多尺度训练有两种实现方式：
+
+1. 在 `train_pipeline` 中输出的每张图都是不定尺度的，然后在 [DataPreprocessor](https://github.com/open-mmlab/mmdetection/blob/3.x/mmdet/models/data_preprocessors/data_preprocessor.py) 中将不同尺度的输入图片
+   通过 [stack_batch](https://github.com/open-mmlab/mmengine/blob/main/mmengine/model/base_model/data_preprocessor.py#L260) 函数填充到同一尺度，从而组成 batch 进行训练。MMDet 中大部分算法都是采用这个实现方式。
```
Comment (PeterH0323) [inline docs/zh_cn/common_usage/ms_training_testing.md:8]: 这个换行是需要的吗

File: docs/zh_cn/common_usage/ms_training_testing.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+2. 在 `train_pipeline` 中输出的每张图都是固定尺度的，然后直接在 DataPreprocessor 中进行 batch 张图片的上下采样，从而实现多尺度训练功能
+
+理论上第一种实现方式所生成的尺度会更加丰富，但是由于其对单张图进行独立增强，训练效率不然第二种方式。在 MMYOLO 中两种多尺度训练方式都是支持的，但是我推荐你使用第二种。
+
+以 `configs/yolov5/yolov5_s-v61_fast_1xb12-40e_cat.py` 配置为例，其默认配置采用的是 640x640 固定尺度训练，假设想实现以 32 为倍数，且多尺度范围为 (480, 800) 的训练方式，则可以参考 YOLOX 做法通过 DataPreprocessor 中的 [YOLOXBatchSyncRandomResize](https://github.com/open-mmlab/mmyolo/blob/main/mmyolo/models/data_preprocessors/data_preprocessor.py#L20) 实现。
```
Comment (PeterH0323) [inline docs/zh_cn/common_usage/ms_training_testing.md:13]:
```suggestion
以 `configs/yolov5/yolov5_s-v61_fast_1xb12-40e_cat.py` 配置为例，其默认配置采用的是 640x640 固定尺度训练，假设想实现以 32 为倍数，且多尺度范围为 (480, 800) 的训练方式，则可以参考 YOLOX 做法通过 `DataPreprocessor` 中的 [YOLOXBatchSyncRandomResize](https://github.com/open-mmlab/mmyolo/blob/main/mmyolo/models/data_preprocessors/data_preprocessor.py#L20) 实现。
```

File: docs/zh_cn/common_usage/ms_training_testing.md (hunk 1 above)
Comment (hhaAndroid) [reply docs/zh_cn/common_usage/ms_training_testing.md:8]: 不需要

File: docs/en/common_usage/ms_training_testing.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+2. Each image output in `train_pipeline` is at fixed scale, and then up and down sampling of batches of images directly in `DataPreprocessor` for multi-scale training.
+
+Theoretically, the first implementation generates richer scales, but due to its independent data augmentation of a single image, the training efficiency is not as efficient as the second one. Both multi-scale training methods are supported in MMYOLO, but I recommend you use the second one.
+
+Take `configs/yolov5/yolov5_s-v61_fast_1xb12-40e_cat.py` configuration as an example, its default configuration is 640x640 fixed scale training, suppose you want to implement training in multiples of 32 and multi-scale range (480, 800), you can refer to YOLOX practice by [YOLOXBatchSyncRandomResize](https://github.com/open-mmlab/mmyolo/blob/main/mmyolo/models/data_preprocessors/data_preprocessor.py#L20) in the DataPreprocessor.
```
Comment (RangeKing) [inline docs/en/common_usage/ms_training_testing.md:12]:
```suggestion
Take `configs/yolov5/yolov5_s-v61_fast_1xb12-40e_cat.py` configuration as an example, its default configuration is 640x640 fixed scale training, suppose you want to implement training in multiples of 32 and multi-scale range (480, 800), you can refer to YOLOX practice by [YOLOXBatchSyncRandomResize](https://github.com/open-mmlab/mmyolo/blob/dc85144fab20a970341550794857a2f2f9b11564/mmyolo/models/data_preprocessors/data_preprocessor.py#L20) in the DataPreprocessor.
```

File: docs/zh_cn/common_usage/ms_training_testing.md (hunk 1 above)
Comment (RangeKing) [inline docs/zh_cn/common_usage/ms_training_testing.md:8]:
```suggestion
   通过 [stack_batch](https://github.com/open-mmlab/mmengine/blob/dbae83c52fa54d6dda08b6692b124217fe3b2135/mmengine/model/base_model/data_preprocessor.py#L260-L261) 函数填充到同一尺度，从而组成 batch 进行训练。MMDet 中大部分算法都是采用这个实现方式。
```

File: docs/zh_cn/common_usage/ms_training_testing.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+MMYOLO 中目前支持了主流的 YOLOv5、YOLOv6、YOLOv7、YOLOv8 和 RTMDet 等算法，其默认配置均为单尺度 640x640 训练。 在 MM 系列开源库中常用的多尺度训练有两种实现方式：
+
+1. 在 `train_pipeline` 中输出的每张图都是不定尺度的，然后在 [DataPreprocessor](https://github.com/open-mmlab/mmdetection/blob/3.x/mmdet/models/data_preprocessors/data_preprocessor.py) 中将不同尺度的输入图片
+   通过 [stack_batch](https://github.com/open-mmlab/mmengine/blob/main/mmengine/model/base_model/data_preprocessor.py#L260) 函数填充到同一尺度，从而组成 batch 进行训练。MMDet 中大部分算法都是采用这个实现方式。
+2. 在 `train_pipeline` 中输出的每张图都是固定尺度的，然后直接在 `DataPreprocessor` 中进行 batch 张图片的上下采样，从而实现多尺度训练功能
+
+理论上第一种实现方式所生成的尺度会更加丰富，但是由于其对单张图进行独立增强，训练效率不然第二种方式。在 MMYOLO 中两种多尺度训练方式都是支持的，但是我推荐你使用第二种。
+
+以 `configs/yolov5/yolov5_s-v61_fast_1xb12-40e_cat.py` 配置为例，其默认配置采用的是 640x640 固定尺度训练，假设想实现以 32 为倍数，且多尺度范围为 (480, 800) 的训练方式，则可以参考 YOLOX 做法通过 DataPreprocessor 中的 [YOLOXBatchSyncRandomResize](https://github.com/open-mmlab/mmyolo/blob/main/mmyolo/models/data_preprocessors/data_preprocessor.py#L20) 实现。
```
Comment (RangeKing) [inline docs/zh_cn/common_usage/ms_training_testing.md:13]:
```suggestion
以 `configs/yolov5/yolov5_s-v61_fast_1xb12-40e_cat.py` 配置为例，其默认配置采用的是 640x640 固定尺度训练，假设想实现以 32 为倍数，且多尺度范围为 (480, 800) 的训练方式，则可以参考 YOLOX 做法通过 `DataPreprocessor` 中的 [YOLOXBatchSyncRandomResize](https://github.com/open-mmlab/mmyolo/blob/dc85144fab20a970341550794857a2f2f9b11564/mmyolo/models/data_preprocessors/data_preprocessor.py#L20) 实现。
```

File: docs/en/common_usage/ms_training_testing.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
+# Multi-scale training and testing
+
+## Multi-scale training
+
+The popular YOLOv5, YOLOv6, YOLOv7, YOLOv8 and RTMDet algorithms are supported in MMYOLO currently, and their default configuration is single-scale 640x640 training. There are two implementations of multiscale training commonly used in the MM family of open source libraries
+
+1. Each image output in `train_pipeline` is at variable scale, and pad different scales of input images to the same scale by [stack_batch](https://github.com/open-mmlab/mmengine/blob/main/mmengine/model/base_model/data_preprocessor.py#L260) function in [DataPreprocessor](https://github.com/open-mmlab/mmdetection/blob/3.x/mmdet/models/data_preprocessors/data_preprocessor.py). Most of the algorithms in MMDet are implemented using this approach.
+2. Each image output in `train_pipeline` is at fixed scale, and then up and down sampling of batches of images directly in `DataPreprocessor` for multi-scale training.
```
Comment (RangeKing) [inline docs/en/common_usage/ms_training_testing.md:8]:
```suggestion
2. Each image output in `train_pipeline` is at a fixed scale, and `DataPreprocessor` performs up- and down-sampling of image batches for multi-scale training directly.
```

File: docs/zh_cn/common_usage/ms_training_testing.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+## 多尺度训练
+
+MMYOLO 中目前支持了主流的 YOLOv5、YOLOv6、YOLOv7、YOLOv8 和 RTMDet 等算法，其默认配置均为单尺度 640x640 训练。 在 MM 系列开源库中常用的多尺度训练有两种实现方式：
+
+1. 在 `train_pipeline` 中输出的每张图都是不定尺度的，然后在 [DataPreprocessor](https://github.com/open-mmlab/mmdetection/blob/3.x/mmdet/models/data_preprocessors/data_preprocessor.py) 中将不同尺度的输入图片
+   通过 [stack_batch](https://github.com/open-mmlab/mmengine/blob/main/mmengine/model/base_model/data_preprocessor.py#L260) 函数填充到同一尺度，从而组成 batch 进行训练。MMDet 中大部分算法都是采用这个实现方式。
+2. 在 `train_pipeline` 中输出的每张图都是固定尺度的，然后直接在 `DataPreprocessor` 中进行 batch 张图片的上下采样，从而实现多尺度训练功能
+
+理论上第一种实现方式所生成的尺度会更加丰富，但是由于其对单张图进行独立增强，训练效率不然第二种方式。在 MMYOLO 中两种多尺度训练方式都是支持的，但是我推荐你使用第二种。
```
Comment (RangeKing) [inline docs/zh_cn/common_usage/ms_training_testing.md:11]:
```suggestion
在 MMYOLO 中两种多尺度训练方式都是支持的。理论上第一种实现方式所生成的尺度会更加丰富，但是由于其对单张图进行独立增强，训练效率不如第二种方式。所以我们更推荐使用第二种方式。
```

File: docs/zh_cn/common_usage/ms_training_testing.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+                random_size_range=(480, 800),
+                # 输出尺度需要被 32 整除
+                size_divisor=32,
+                # 每隔 1 个迭代改变一次输出输出
+                interval=1)
+        ])
+)
+```
+
+上述配置就可以实现多尺度训练了。为了方便我们已经在 `configs/yolov5/` 下已经提供了该配置。其余 YOLO 系列算法也是类似做法。
```
Comment (RangeKing) [inline docs/zh_cn/common_usage/ms_training_testing.md:37]:
```suggestion
上述配置就可以实现多尺度训练了。为了方便，我们已经在 `configs/yolov5/` 下已经提供了该配置。其余 YOLO 系列算法也是类似做法。
```

File: docs/en/common_usage/ms_training_testing.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+Theoretically, the first implementation generates richer scales, but due to its independent data augmentation of a single image, the training efficiency is not as efficient as the second one. Both multi-scale training methods are supported in MMYOLO, but I recommend you use the second one.
```
Comment (RangeKing) [inline docs/en/common_usage/ms_training_testing.md:10]:
```suggestion
Both two multi-scale training approaches are supported in MMYOLO. Theoretically, the first implementation can generate richer scales, but its training efficiency is not as good as the second one due to its independent augmentation of a single image. Therefore, we recommend using the second approach.
```

File: docs/en/common_usage/ms_training_testing.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+                # multi-scale range (480, 800)
+                random_size_range=(480, 800),
+                # The output scale needs to be divisible by 32
+                size_divisor=32,
+                interval=1)
+        ])
+)
+```
+
+The above configuration will enable multi-scale training. We have already provided this configuration under `configs/yolov5/` for convenience. The rest of the YOLO family of algorithms is similar.
```
Comment (RangeKing) [inline docs/en/common_usage/ms_training_testing.md:35]:
```suggestion
The above configuration will enable multi-scale training. We have already provided this configuration under `configs/yolov5/` for convenience. The rest of the YOLO family of algorithms are similar.
```

**Pull request merged**